### PR TITLE
Write documentation

### PR DESCRIPTION
**Pull request by MarcoLm993**
Please let me know if you see anything missing :)

### CODE REVIEW

File: include/model_checker/state_generation.h [hunk 1]
```diff
@@ -82,22 +103,40 @@ class StateGeneration {
...
+     * @brief Getter for the min amount of steps to compute before checking for target and condition formulae
+     * @return The lower bound of the property
+     */
     inline size_t getLowerBound() const
     {
         return _property_description.getLowerBound();
     }
 
+    /*!
+     * @brief Getter for the max amount of steps, before which the target formula must hold to be verified
```
Comment (MKlauck) [inline include/model_checker/state_generation.h:128]: suggestion: ... before the target formula to be verified must hold.

File: include/model_checker/statistical_model_checking_engine.hpp [hunk 2]
```diff
@@ -55,22 +55,56 @@ namespace model_checker {
...
 
+    /*!
+     * @brief Check if the provided property is supported by the engine
+     * @param check_task The property to verify
+     * @return true if the property can be handled, false otherwise
+     */
     virtual bool canHandle(storm::modelchecker::CheckTask<storm::logic::Formula, ValueType> const& check_task) const override;
 
+    /*!
+     * @brief Static version of the engine support check for the provided property
```
Comment (MKlauck) [inline include/model_checker/statistical_model_checking_engine.hpp:84]: supports?

File: include/samples/batch_results.hpp [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    /*!
+     * @brief BatchResults constructor
+     * @param batch_size The predefined size of each batch
+     * @param prop The property type we are evaluating (probability or reward), to determine what information to store
+     */
+    BatchResults(size_t batch_size, const state_properties::PropertyType prop);
+    
+    /*!
+    * @brief Check whether we need more samples according to the batch size
+    * @return true if the have less elements than the batch size. False otherwise
```
Comment (MKlauck) [inline include/samples/batch_results.hpp:66]: they

**Review by MKlauck — APPROVED** [review]

File: docs/source/installation.rst [hunk 4]
```diff
@@ -0,0 +1,48 @@
+Installation Guide
+==================
+
+How to install smc_storm
+------------------------
+
+Install STORM
++++++++++++++
+
+smc_storm needs STORM to be built on the local machine. To achieve that, follow the `official documentation <https://www.stormchecker.org/documentation/obtain-storm/build.html>`_.
```
[the remark this answers is not in the document]
Comment (MarcoLm993) [reply docs/source/installation.rst:10]: Good point, I will adapt it

File: README.md
[the remark this answers is not in the document]
Comment (MarcoLm993) [reply README.md:1]: I think it is a matter of taste, but I agree that it is one less thing to maintain :)

File: include/model_checker/statistical_model_checking_engine.hpp (hunk 2 above)
Comment (MarcoLm993) [reply include/model_checker/statistical_model_checking_engine.hpp:84]: It was a "support check", guess can be changed to compatibility check

File: include/samples/batch_results.hpp (hunk 3 above)
Comment (MarcoLm993) [reply include/samples/batch_results.hpp:66]: Not sure how I came up with that strange sentence... Will reformulate

File: README.md
[the remark this answers is not in the document]
Comment (MarcoLm993) [reply README.md:1]: By the way, I will change the ReadMe at a second point in time, to make it similar to the one from the mc_toolchain_jani package (need to check how it looks like with the GPL license, but probably it is almost the same as for Apache)

File: include/model_checker/state_generation.h [hunk 5]
```diff
@@ -82,22 +103,40 @@ class StateGeneration {
 
     std::size_t getNumberOfInitialStates() const;
 
-    // Check whether the loaded state satisfies the condition formula
+    /*!
+     * @brief Check whether the loaded state satisfies the condition formula
+     * @return true of the condition formula is satisfied, false otherwise
```
[the remark this answers is not in the document]
Comment (MKlauck) [reply include/model_checker/state_generation.h:108]: Actually, I meant the sentence in line 108: true if the condition...

File: include/model_checker/state_generation.h [hunk 6]
```diff
@@ -82,22 +103,40 @@ class StateGeneration {
...
+     */
     bool isConditionState() const;
 
-    // Check whether the loaded state satisfies the target formula
+    /*!
+     * @brief Check whether the loaded state satisfies the target formula
+     * @return true of the target formula is satisfied, false otherwise
```
[the remark this answers is not in the document]
Comment (MKlauck) [reply include/model_checker/state_generation.h:114]: Same as above

**Review by MKlauck — COMMENTED** [review]
Is there a reason why there is no yaml file to automatically build the documentation when pushing?

**Pull request merged**